File: C++/137. Single Number II.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        unordered_map<int,int> map;
        for (int i = 0; i<nums.size(); i++){
            if (map.count(nums[i])>0){
                map[nums[i]]++;
            }
            else {
                map[nums[i]] = 1;
            }
        }
        for (int i = 0; i<nums.size(); i++){
            if(map[nums[i]]==1){
                return nums[i];
            }
        }
        return 0;
    }
};

File: C#/137. Single Number II.cs
public class Solution {
    public int SingleNumber(int[] nums) {
        Dictionary<int, int> dict = new Dictionary<int, int>();
        for (int i = 0; i<nums.Length; i++){
            if (dict.ContainsKey(nums[i])){
                dict[nums[i]]++;
            }
            else {
                dict[nums[i]]=1;
            }
        }
        for (int i = 0; i<nums.Length; i++){
            if(dict[nums[i]]==1){
                return nums[i];
            }
        }
        return 0;
    }
}
